vdirsyncer/storage/google.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+import logging
+
+import click
+
+from . import dav
+from .. import exceptions, utils
+
+logger = logging.getLogger(__name__)
+
+
+TOKEN_URL = 'https://accounts.google.com/o/oauth2/v2/auth'
+REFRESH_URL = 'https://www.googleapis.com/oauth2/v4/token'
+
+CLIENT_ID = '155040592229-mth5eiq7qt9dtk46j0vcnoometuab9mb.apps.googleusercontent.com'
+CLIENT_SECRET = 'flVXF7jB2A2-YC4rg2sUCCX1'
+
+try:
+    from requests_oauthlib import OAuth2Session
+    have_oauth2 = True
+except ImportError:
+    have_oauth2 = False
+
+
+class GoogleSession(dav.DavSession):
+    def __init__(self, refresh_token=None, client_id=CLIENT_ID,
+                 client_secret=CLIENT_SECRET):
+        self.useragent = client_id
+        self._settings = {}
+
+        if not have_oauth2:
+            raise exceptions.UserError('requests-oauthlib not installed')
+
+        if refresh_token:
+            token = {
+                'refresh_token': refresh_token,
+                'access_token': 'dummy',
```

Comment:
So I'm unsure whether this part works as it should, and I'm now unsure whether just requesting a new access token was a good idea after all.